src/FixedDataTable.js

```diff
@@ -598,7 +598,7 @@ var FixedDataTable = createReactClass({
 
     // Cancel any pending debounced scroll handling and handle immediately.
     this._didScrollStop.reset();
-    this._didScrollStopSync();
+    this._didScrollStop();
```

Comment:
can we just remove the scroll calls alltogether?

Reply:
Also, If we manually call _didScrollStop, won't it just debounce it again, and delay it? If the user keeps on changing props, wouldn't this lead this to didScrollStop never being called?

Reply:
@pradeepnschrodinger probably no reason to manually stop scrolling, but `onScrollEnd` needs to be called, and that's only happening in `didScrollStop`. tried to remove those lines but that indeed broke some behaviour in our use-case (because `onScrollEnd` was not called).

Reply:
and i guess it needs to be debounced so that it only gets called after the scrolling has actually stopped, not on every "tick".

Reply:
> @pradeepnschrodinger probably no reason to manually stop scrolling, but `onScrollEnd` needs to be called, and that's only happening in `didScrollStop`. tried to remove those lines but that indeed broke some behaviour in our use-case (because `onScrollEnd` was not called).

Sorry, just to be sure: have you tried also deleting both
```
    this._didScrollStop.reset();
    this._didScrollStopSync();
```
For me, onScrollEnd is called, as expected because the debouncer will eventually call didScrollStopSync.

Reply:
actually the problem is that if i remove those two lines, the first row and header row are not getting re-rendered when props change (i guess `children` in this case). we are showing loading spinners in the header and in the first row while loading, the spinner in the first row only goes away when scrolled away (gets removed from the DOM) and scrolling back to it.

Reply:
Not sure if I got you (maybe we might need a minimal working demo for this...). 
The rows are rendered with the `isScrolling` prop, the expected behavior should be that the rows (including the header) will be re-rendered when scrolling is stopped. 
But `FixedDataTableCell` [won't update](https://github.com/schrodinger/fixed-data-table-2/blob/master/src/FixedDataTableCell.js#L103) if scrolling is on AND the row index hasn't changed (for performance concerns). Maybe that's your issue? It could be that for some reason the scrolling state is ON throughout and hence the cell gets updated only on row index change (which is what happens when it gets removed from the DOM or added back in).

@bodyflex Can you check if the following solves the issue? Move the this.didScrollStop() call under the componentWillRecieveProps to inside the controlled scrolling logic. Something like:
```
  componentWillReceiveProps(/*object*/ nextProps) {
    var newOverflowX = nextProps.overflowX;
    var newOverflowY = nextProps.overflowY;

    // In the case of controlled scrolling, notify.
    if (this.props.ownerHeight !== nextProps.ownerHeight ||
        this.props.scrollTop !== nextProps.scrollTop ||
        this.props.scrollLeft !== nextProps.scrollLeft) {
      this._didScrollStart();
      this._didScrollStop();  // ADDED
    }
    // this._didScrollStop(); // REMOVED

    this.setState(this._calculateState(nextProps, this.state));
  },
```
and report back what you see.

Thanks for the follow ups :)

Reply:
yeah sorry, it's hard to explain in words and constructing a minimal working demo from our very complex use-case would be hard.

anyway, i tried the modification you suggested, and that does seem to work: the performance is good, and the rows are getting re-rendered. would you prefer that; should i commit the change?

Reply:
@bodyflex Oh, that's fine then. Thanks for the help.

So probably what happened was controlled scrolling occurs in your app. But our code has a bug in that it only calls `_didScrollStart` to start scrolling, but never calls `_didScrollStop` to end scrolling. Hence since the `isScrolling` is always turned on, the cells didn't get re-rendered.

Yes, thanks. Please remove the scrollStop calls outside of controlled scrolling logic, and keep a _didScrollStop inside. 

```
componentWillReceiveProps(/*object*/ nextProps) {
    var newOverflowX = nextProps.overflowX;
    var newOverflowY = nextProps.overflowY;

    // In the case of controlled scrolling, notify.
    if (this.props.ownerHeight !== nextProps.ownerHeight ||
        this.props.scrollTop !== nextProps.scrollTop ||
        this.props.scrollLeft !== nextProps.scrollLeft) {
      this._didScrollStart();
      this._didScrollStop();
    }

    this.setState(this._calculateState(nextProps, this.state));
},
```

And this PR should be good to getting merged :)

Reply:
great, done!